src/RSocketClient.cpp

```diff
@@ -41,11 +44,11 @@ folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
     keepaliveTimer = std::move(keepaliveTimer),
     stats = std::move(stats),
     networkStats = std::move(networkStats),
-    promise = std::move(promise)](
-      std::unique_ptr<DuplexConnection> connection,
-      folly::EventBase& eventBase) mutable {
+    promise = std::move(promise)
+  ](std::unique_ptr<DuplexConnection> connection,
+    folly::EventBase & eventBase) mutable {
     VLOG(3) << "onConnect received DuplexConnection";
-
+    evb_ = &eventBase;
```

Comment:
The RSocketClient just keeps track of the evb which the client-connection is using, so that later it can use it for scheduling disconnect in the right thread and to ensure resumption is happening in the same evb-thread.